tools/release.rb

```diff
@@ -23,96 +23,68 @@
 require_relative "spec_template"
 
 class Release
-  def self.version
-    if File.exist?("version.go")
-      File.read("version.go")[/(\d+\.\d+\.\d+)/]
-    else
-      Machinery::VERSION
-    end
-  end
+  include ReleaseChecks
+
+  RPM_CHANGES_FILE = File.join(Machinery::ROOT, "package/machinery.changes")
+  CHANGELOG_FILES = [
+    File.join(Machinery::ROOT, "NEWS"),
+    File.join(Machinery::ROOT, "RPM_CHANGES")
+  ]
 
   def initialize(opts = {})
-    @current_version = Release.version
     @options = {
-      version:      Release.generate_development_version,
-      skip_rpm_cleanup: false,
-      jenkins_name:     "machinery-unit",
-      changes_file:     "NEWS",
-      rpm_changes_file: "RPM_CHANGES",
-      package_name:     "machinery"
+      version: generate_development_version,
+      skip_rpm_cleanup: false
     }.merge(opts)
-    @tag             = "v#{@options[:version]}"
-    @release_time    = Time.now.strftime('%a %b %d %H:%M:%S %Z %Y')
-    @mail            = Cheetah.run(["git", "config", "user.email"], :stdout => :capture).chomp
-    @gemspec         = Gem::Specification.load("machinery.gemspec")
-    @release_checks  = ReleaseChecks.new(@tag, @options[:jenkins_name])
-    @changelog_files = [@options[:changes_file], @options[:rpm_changes_file]]
-    @package_changes_path = "package/#{@options[:package_name]}.changes"
-    set_version
-    add_default_rpm_changes_entry
-  end
-
-  def check
-    @release_checks.check
+    @release_version = @options[:version]
```

Comment:
As you store the version in the `@release_version` variable, you should use that all over the place and not access @options[:version] in other places of the code.

It also would be better to have a method to access it, rather than using the member variable directly. This is clearer and more robust.